### PR TITLE
chore: upgrade React to v17

### LINKED ISSUE #44: Support React 17
The package still depends on react ^16.8.3, it would be great to support React 17 in order to avoid necessary resolutions on the user side to avoid potential https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react

### PR DESCRIPTION
**Pull request by mcmxcdev**
Closes #44.

## Further explanation 
The React upgrade itself would have been done with just upgrading the `react` and `react-dom` packages, but one of the great new features is: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform

Both `babel.config.js` and `tsconfig.json` are now updated to use the new JSX transform.

You can read more about React 17 migration here: https://reactjs.org/blog/2020/10/20/react-v17.html

## Additional improvements

Not part of the scope but additionally, I renamed the `check` scripts to `typecheck` in package.json because check is already a reserved command in yarn for `yarn check`.

I also had to upgrade Typescript to at least v4.1. to be able to use the new JSX transform feature.

Also put storybook-static folder into `.gitignore`, fixed deprecated storybook glob pattern in `main.js` and ignored an eslint issue in `0-Blog.stories.js`

### CODE REVIEW

**Comment by mhatvan** [comment]
@amccloud Ready for review!

File: .storybook/main.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module.exports = {
-  stories: ["../examples/**/*.stories.(js|md)"],
+  stories: ["../examples/**/*.stories.@(js|md)"],
```
Comment (amccloud) [inline .storybook/main.js:2]: What is the @ for?

**Review by amccloud — COMMENTED** [review]
Thanks for all the updates! Just one question and I think this is good to go

File: .storybook/main.js (hunk 1 above)
Comment (mhatvan) [reply .storybook/main.js:2]: You can read about it here: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#correct-globs-in-mainjs

**Comment by mhatvan** [comment]
@amccloud Great to see this merged! Can we get a new release for this?

**Comment by amccloud** [comment]
@mhatvan can do. I was testing the release out on our internal storybook and ran into an unrelated hiccup that took up my time. I’m ooo til next week and can get a release cut then

**Comment by mhatvan** [comment]
Ok great to hear, looking forward to it!

**Comment by amccloud** [comment]
`6.2.0` has been released

**Pull request merged**